packages/blocks/src/api/raw-handling/ms-newline-normaliser.js

```diff
@@ -0,0 +1,11 @@
+export default function( node ) {
+	if ( node.nodeName !== 'P' ) {
+		return;
+	}
+
+	if ( ! [ ...node.classList ].find( ( className ) => className.match( /^Mso[A-Z].*/ ) ) ) {
```

Comment:
Do you think this is a unique MS Word problem? More specifically a paragraph problem? In my opinion, we should be writing a general line break (`\n`) collapser, as line breaks (`\n`) in HTML are not visible in any circumstance. The are only used to format the HTML.

Reply:
There are, for sure, other tools that do weird things with line breaks. However, this PR (and the related issue) is specific to pasting content from MS Word, where one gets additional line breaks that have been injected by MS Word itself.

It might be true that HTML, or rather the browser, doesn't care about newlines. But they will also end up in `content.raw` in the REST API response. And this is where things can break.

Reply:
Yes, there are other sources that may inject line breaks. That's why I'm suggesting to look for a more general fix. See #17470.